test/rekt/features/apiserversource/data_plane.go

```diff
@@ -766,3 +766,58 @@ func SendsEventsWithRetries() *feature.Feature {
 			).AtLeast(1))
 	return f
 }
+
+func SendsEventsWithBrokerAsSinkTLS() *feature.Feature {
+	src := feature.MakeRandomK8sName("apiserversource")
+	brokerName := feature.MakeRandomK8sName("broker")
+	sacmName := feature.MakeRandomK8sName("apiserversource")
+	sink := feature.MakeRandomK8sName("sink")
+	via := feature.MakeRandomK8sName("via")
+	f := feature.NewFeature()
```

Comment:
A prerequisite of this test is that the transport-encryption is set to strict, see https://github.com/knative/eventing/pull/7103 on how to configure the feature with such prerequisite